superset/migrations/versions/1d73b15530e7_update_tables.py

```diff
@@ -0,0 +1,93 @@
+"""update tables
+
+Revision ID: 1d73b15530e7
+Revises: 1d9e835a84f9
+Create Date: 2018-07-20 11:36:04.535859
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '1d73b15530e7'
+down_revision = '1d9e835a84f9'
+
+from alembic import op
+import sqlalchemy as sa
+
+from superset.utils import generic_find_uq_constraint_name
+
+
+conv = {
+    'uq': 'uq_%(table_name)s_%(column_0_name)s',
+}
+
+tables = sa.Table(
+    'tables',
+    sa.MetaData(),
+    sa.Column('id', sa.Integer, primary_key=True),
+    sa.Column('database_id', sa.Integer),
+    sa.Column('schema', sa.String(127)),
+    sa.Column('table_name', sa.String(127), nullable=False),
+)
+
+
+def upgrade():
+    bind = op.get_bind()
+    insp = sa.engine.reflection.Inspector.from_engine(bind)
+
+    # Reduce the size of the schema and table_name columns for constraint
+    # viability. Note most engines have a 64 character limit for these fields.
+    with op.batch_alter_table('tables') as batch_op:
+        batch_op.alter_column(
+            'schema',
+            existing_nullable=True,
+            existing_type=sa.String(255),
+            type_=sa.String(127),
+        )
+
+        # Additionally enforce that the table_name column be non-nullable.
+        batch_op.alter_column(
+            'table_name',
+            existing_type=sa.String(250),
+            nullable=False,
+            type_=sa.String(127),
+        )
+
+    # Add the missing uniqueness constraint.
+    with op.batch_alter_table('tables', naming_convention=conv) as batch_op:
+        batch_op.create_unique_constraint(
+            'uq_tables_database_id',
+            ['database_id', 'schema', 'table_name'],
+        )
+
+
+def downgrade():
+    bind = op.get_bind()
+    insp = sa.engine.reflection.Inspector.from_engine(bind)
+
+    # Drop the missing uniqueness constraint.
+    with op.batch_alter_table('tables', naming_convention=conv) as batch_op:
+        batch_op.drop_constraint(
+            generic_find_uq_constraint_name(
+                'tables',
+                {'database_id', 'schema', 'table_name'},
```

Comment:
One of the issues with this schema uniqueness is the pesky 'sql' column on the table: Effectively we would ideally like to have two different views of the same table, both with different sql but with the same <database, schema, table_name>.

Reply:
I think the issue here is that if there were two different views of the same table how would one differentiate between them as datasources. Note in a database construct view names need to be unique.

Reply:
I didn't mean real 'SQL views'. I meant basically having the same table added twice, but with different 'sql' like so:

Table foo with sql: 'select foo.a, foo.b, count(1) from foo group by a, b'. 
And another with Table foo with sql: 'select foo.*, some_function() from foo'

Reply:
Personally they feel like different views and thus should be represented as different entities in the database table. If they're both called `foo` it's not apparent to the user which one the datasource is referring to and thus this would lead to confusion.

Reply:
@mistercrunch [here's](https://stackoverflow.com/a/35851332) a suggestion for increasing the maximum column size. The issue is this requires an engine configuration change which I assume is non-ideal.